lib/api/environment.rb

```diff
@@ -1,7 +1,30 @@
 module Api
   class Environment
     def self.normalized_attributes
-      @normalized_attributes ||= {:time => {}, :url => {}, :resource => {}, :encrypted => {}}
+      @normalized_attributes ||= {
+        :time      => time_attributes.each_with_object({}) { |attr, hsh| hsh[attr] = true },
```

Comment:
@Fryguy I thought the same thing, but since hash lookup was faster I assumed it was there for that reason. Could we sacrifice a small performance gain for a more appropriate data structure?